### PR TITLE
Add exclude list

### PR DESCRIPTION
**Pull request by leshikus**
I've also cleaned up exit code checks, they are no longer needed due to `-e` setting

I was not able to specify both `.py` file and test case name in the exclude file.

The list of `test_reduce_layouts` failed test cases differs from time to time.

### CODE REVIEW

**Comment by leshikus** [comment]
temp close

File: python/test/exclude_list.conda [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# FIXME https://github.com/intel/intel-xpu-backend-for-triton/issues/806
+test_reduce_layouts
+# FIXME https://github.com/intel/intel-xpu-backend-for-triton/issues/807
+language/test_random.py
```
Comment (pbchekin) [inline python/test/exclude_list.conda:4]: You exclude too much, why not individual tests?

Comment (pbchekin) [reply python/test/exclude_list.conda:4]: `python/test` is not the best place for the exclude list, what about keeping it in `scripts`?

File: scripts/test-triton.sh [hunk 2]
```diff
@@ -109,37 +99,23 @@ run_core_tests() {
   fi
   cd ${CORE_TEST_DIR}
 
-  TRITON_DISABLE_LINE_INFO=1 python3 -m pytest -n 8 --verbose --device xpu language/ --ignore=language/test_line_info.py
-  if [ $? -ne 0 ]; then
-    echo "FAILED: return code $?" ; exit $?
+  exclude_list=/
+  if [ -n ../exclude_list ]; then
```
Comment (pbchekin) [inline scripts/test-triton.sh:103]: Typo?
```suggestion
  if [ -f ../exclude_list ]; then
```

File: scripts/test-triton.sh [hunk 3]
```diff
@@ -109,37 +99,23 @@ run_core_tests() {
   fi
   cd ${CORE_TEST_DIR}
 
-  TRITON_DISABLE_LINE_INFO=1 python3 -m pytest -n 8 --verbose --device xpu language/ --ignore=language/test_line_info.py
-  if [ $? -ne 0 ]; then
-    echo "FAILED: return code $?" ; exit $?
+  exclude_list=/
```
Comment (pbchekin) [inline scripts/test-triton.sh:102]: Have you tested that `pytest ... -k /` works when there is no exclude list?

**Comment by pbchekin** [comment]
See also #810

File: python/test/exclude_list.conda (hunk 1 above)
Comment (leshikus) [reply python/test/exclude_list.conda:4]: >> why not individual tests?
the exact set of failed tests differs from launch to launch, sometimes ~10 tests fail, sometimes ~50

File: scripts/test-triton.sh (hunk 2 above)
Comment (leshikus) [reply scripts/test-triton.sh:103]: Thanks! Fixed

File: scripts/test-triton.sh (hunk 3 above)
Comment (leshikus) [reply scripts/test-triton.sh:102]: I've tested that in a separate branch (and discovered the `-n` typo). I've implemented better solution in the last patch.

File: python/test/exclude_list.conda (hunk 1 above)
Comment (leshikus) [reply python/test/exclude_list.conda:4]: I've moved the file to `/scripts/`

**Comment by pbchekin** [comment]
Please rebase and resolve conflicts.

File: scripts/test-triton.sh [hunk 4]
```diff
@@ -99,7 +99,16 @@ run_core_tests() {
...
+  exclude_list_k=
+  exclude_list=
+  if [ -f ../../../scripts/core_exclude_list ]; then
+    exclude_list_k=-k
+    exclude_list=$(awk -vconj=not -vORS= '/^#/ { next } { print conj" "$1; conj=" and not" }' ../../../scripts/core_exclude_list)
+  fi
+
+  TRITON_DISABLE_LINE_INFO=1 python3 -m pytest -n 8 --verbose --device xpu language/ "$exclude_list_k" "$exclude_list" --ignore=language/test_line_info.py
+
+  TRITON_DISABLE_LINE_INFO=1 python3 -m pytest -n 8 --verbose --device xpu language/ "$exclude_list_k" "$exclude_list" --ignore=language/test_line_info.py
```
Comment (pbchekin) [inline scripts/test-triton.sh:111]: Duplicate?

**Comment by pbchekin** [comment]
Could you please consider using https://pypi.org/project/pytest-select/? The benefit of `pytest-select` is it has `--select-fail-on-missing`. We also can move (gradually, some of) currently skipped tests to the skip files. In this case, conda workflow can specify its own set of files with tests to skip.

File: scripts/test-triton.sh [hunk 5]
```diff
@@ -131,6 +140,7 @@ run_tutorial_test() {
   echo
   echo "****** Running $1 test ******"
   echo
+
```
Comment (pbchekin) [inline scripts/test-triton.sh:143]: This change is not  needed.

**Comment by pbchekin** [comment]
> Could you please consider using https://pypi.org/project/pytest-select/? The benefit of `pytest-select` is it has `--select-fail-on-missing`. We also can move (gradually, some of) currently skipped tests to the skip files. In this case, conda workflow can specify its own set of files with tests to skip.

May be in separate PR, if this one is required for green conda workflow.

**Comment by leshikus** [comment]
migrating to [/pytest-select/](https://pypi.org/project/pytest-select/)?

**Comment by leshikus** [comment]
Note, [pytest-select/](https://pypi.org/project/pytest-select/) requires exact test names for each exclude, https://github.com/ulope/pytest-select/blob/31e39c5a9b08d4509520941d47fe69cbd2d34dbc/pytest_select/plugin.py#L76

**Pull request merged**